aten/src/ATen/MetaInit.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <ATen/MetaInit.h>
+
+#include <cstdint>
+
+#include <ATen/Tensor.h>
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <c10/core/DispatchKeySet.h>
+#include <torch/library.h>
+
+namespace at {
+namespace {
+
+constexpr DispatchKeySet after_meta_init_keyset =
+    DispatchKeySet{DispatchKeySet::FULL_AFTER, DispatchKey::MetaInit};
+
+void metaInitFallback(const OperatorHandle& op, DispatchKeySet ks, torch::jit::Stack* s) {
+  DisableMetaInitGuard guard{};
+
+  // Just redispatch, effectively noop.
+  op.redispatchBoxed(ks & after_meta_init_keyset, s);
+}
+
+// Used to support nested calls.
+thread_local std::size_t meta_init_level = 0;
+
+} // namespace
+
+void enableMetaInit(bool value) {
+  if (value) {
+    meta_init_level++;
+
+    if (meta_init_level == 1) {
+      c10::impl::tls_set_dispatch_key_included(DispatchKey::MetaInit, true);
```

Comment:
Using this strategy still means that something like `torch.tensor` may not be captured by the PyTorch dispatcher (we used this strategy in functorch and we had some special workarounds for torch.tensor). However, it does fix your problem of wanting to see all pytorch operations (especially the ones that happen before autograd)

Reply:
I will release the third PR of this stack later today. There you will see the "surrogate" tensor approach that I used. In fact for meta-init the `torch.tensor` is just one case. Any tensor that was lexically constructed before the context and then later used inside the context has the same issue.